File: src/my_cpp_pkg/src/number_publisher.cpp
#include "rclcpp/rclcpp.hpp"
#include "example_interfaces/msg/int64.hpp"

class NumberPubNode : public rclcpp::Node
{
public:
    NumberPubNode() : Node("number_publisher")
    {
        this->declare_parameter("number_to_publish", 2);
        this->declare_parameter("publish_rate", 1.0);

        number_ = this->get_parameter("number_to_publish").as_int();
        double publish_rate = this->get_parameter("publish_rate").as_double();

        publisher_ = this->create_publisher<example_interfaces::msg::Int64>("number", 10);
        timer_ = this->create_wall_timer(std::chrono::milliseconds((int) (1000.0 / publish_rate)),
                                         std::bind(&NumberPubNode::timerCallback, this));
        RCLCPP_INFO(this->get_logger(), "number_publisher has been started");
    }

private:
    void timerCallback()
    {
        auto msg = example_interfaces::msg::Int64();
        msg.data = number_;
        publisher_->publish(msg);
    }

    int number_;
    rclcpp::Publisher<example_interfaces::msg::Int64>::SharedPtr publisher_;
    rclcpp::TimerBase::SharedPtr timer_;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberPubNode>();
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: src/my_py_pkg/my_py_pkg/number_publisher.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from example_interfaces.msg import Int64

class NumberPublisherNode(Node):
    def __init__(self):
        super().__init__("number_publisher")
        self.declare_parameter("number_to_publish", 2)
        self.declare_parameter("publish_rate", 1.0)
        
        
        self.number = self.get_parameter("number_to_publish").value
        self.publish_rate = self.get_parameter("publish_rate").value
        
        self.publisher_ = self.create_publisher(Int64,"number",10)
        timer_ = self.create_timer(1.0 / self.publish_rate ,self.timerCallback)
        self.get_logger().info("number_publisher has been started")

    def timerCallback(self):
        msg = Int64()
        msg.data = self.number
        self.publisher_.publish(msg)

def main(args=None):
    rclpy.init(args=args)
    node = NumberPublisherNode()
    rclpy.spin(node)
    rclpy.shutdown()

if __name__ == "__main__":
    main()